src/components/ParserComponent.tsx

```diff
@@ -70,15 +72,93 @@ export default class ParserComponent extends React.Component<ParserInterface,
     return true;
   }
 
+  /**
+   * Purpose: to instantiate an empty list of objects
+   * for tracking the kinds of data in a column
+   * @param: fieldLength the number of columns of data
+   * @return: a list of objects
+   */
+  // eslint-disable-next-line max-len,require-jsdoc
+  createTypeCountingObjects(fieldLength: number) : CountTypes[] {
+    const typesForEachCol = [];
+    // instantiate object for each column
+    for (let i = 0; i < fieldLength; i++) {
+      typesForEachCol.push(new CountTypes());
+    }
+    return typesForEachCol;
+  }
+
   /**
    * Purpose: attempts to infer the types of the data in each of the columns
    * of the csv data
-   * @param {Array} data: the array of data to infer the types for
-   * @return {Array}: a list of objects which define the methods available for
-   * the data
+   * @param {Array} data: the array of pre-sorted valid data
+   * @return {Array}: a list of objects of type Column
    */
-  inferTypes(data: Array<object>): Array<object> {
-    return [];
+  inferTypes(data: Array<object>): Array<Column> {
```

Comment:
This *might* trigger a cognitive complexity issue. Maybe to counteract that if we can't find a way to simplify it further, we could provide more detailed docs about how this function is implemented and the steps it takes. It would make it easier for other devs to understand the function (even though this is well inline-commented)

Reply:
Ya sonar reports a cognative complexity of 30(!). We can choose to ignore this or mitigate it. Up to you.

https://sonarcloud.io/project/issues?id=cmpt371-team2&open=AW4bHPFd_lhI3gzXfLi7&pullRequest=70&resolved=false&types=CODE_SMELL

Reply:
I think it just needs to get done, I know it's complex but inferring the type was more challenging than anticipated. If you would like we could go over the code during pair programming to try and simplify it but as I stand I am pretty sick of looking at it.

Reply:
fair enough.